### PR TITLE
Fix bag relationships sort

### PR DESCRIPTION
**Pull request by mshima**
Fix https://github.com/jhipster/generator-jhipster/issues/17990 and fix https://github.com/jhipster/generator-jhipster/issues/18156

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by DanielFran — DISMISSED** [review]

**Comment by DanielFran** [comment]
@vishal423 Can you confirm it is OK for you, please?

**Review by vishal423 — DISMISSED** [review]
It would be good to add Junit tests to assert this custom logic. Also, in my opinion, we shouldn't do eager fetch in getAll apis and it should be optional as it was before.

**Comment by DanielFran** [comment]
@mshima can you add some tests?

File: generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs [hunk 1]
```diff
@@ -78,14 +80,19 @@ public class <%= entityClass %>RepositoryWithBagRelationshipsImpl implements <%=
     }
 
     List<<%= persistClass %>> fetch<%= relationship.relationshipNameCapitalizedPlural %>(List<<%= persistClass %>> <%= entityInstancePlural %>) {
-        return entityManager
+        HashMap<Object, Integer> order = new HashMap<Object, Integer>();
```
Comment (Tcharl) [inline generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs:83]: Could you add the order in a dedicated private method (or at least the sorting stuff at the bottom of the method)

**Comment by pascalgrimaud** [comment]
> Also, in my opinion, we shouldn't do eager fetch in getAll apis and it should be optional as it was before.

totally agree with @vishal423 here

**Comment by DanielFran** [comment]
@mshima, Can you update the PR to not fetch eagerly by default?

File: generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs (hunk 1 above)
Comment (mshima) [reply generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs:83]: Does the createQuery function changes the original array?

**Review by vishal423 — APPROVED** [review]

**Pull request merged**